diff --git a/ctags/main/parse.c b/ctags/main/parse.c
index c9d15a5d..b3bc7a44 100644
--- a/ctags/main/parse.c
+++ b/ctags/main/parse.c
@@ -4777,12 +4777,18 @@ extern void anonGenerate (vString *buffer, const char *prefix, int kind)
 	parser -> anonymousIdentiferId ++;
 
 	char szNum[32];
+#if 0
 	char buf [9];
 
 	vStringCopyS(buffer, prefix);
 
 	anonHashString (getInputFileName(), buf);
 	sprintf(szNum,"%s%02x%02x",buf,parser -> anonymousIdentiferId, kind);
+#else
+	/* we want to see numbers for anon functions in the tree view instead of the hash */
+	vStringCopyS(buffer, prefix);
+	sprintf(szNum,"%u", parser -> anonymousIdentiferId);
+#endif
 	vStringCatS(buffer,szNum);
 }
 
diff --git a/ctags/parsers/kotlin.c b/ctags/parsers/kotlin.c
index 12c4d1e5..21e8998d 100644
--- a/ctags/parsers/kotlin.c
+++ b/ctags/parsers/kotlin.c
@@ -26,7 +26,7 @@ static size_t pcc_strnlen(const char *str, size_t maxlen) {
 #define PCC_DBG_MATCH    1
 #define PCC_DBG_NOMATCH  2
 
-#include "./peg/kotlin.h"
+#include "kotlin.h"
 
 #include "kotlin_pre.h"
 #ifndef PCC_BUFFERSIZE
