diff --git a/lexilla/lexers/LexJulia.cxx b/geany/scintilla/lexers/LexJulia.cxx
index 6730074..ccf947d 100644
--- a/lexilla/lexers/LexJulia.cxx
+++ b/geany/scintilla/lexers/LexJulia.cxx
@@ -39,7 +39,8 @@
 #include "DefaultLexer.h"
 
 using namespace Scintilla;
-using namespace Lexilla;
+// Geany still uses Scintilla v3.5
+//using namespace Lexilla;
 
 static const int MAX_JULIA_IDENT_CHARS = 1023;
 
@@ -138,7 +139,9 @@ public:
 		delete this;
 	}
 	int SCI_METHOD Version() const override {
-		return lvRelease5;
+		// Geany still uses Scintilla v3.5
+		//return lvRelease5;
+		return lvIdentity;
 	}
 	const char * SCI_METHOD PropertyNames() override {
 		return osJulia.PropertyNames();
@@ -163,7 +166,9 @@ public:
 		return 0;
 	}
 
-	static ILexer5 *LexerFactoryJulia() {
+	// Geany still uses Scintilla v3.5
+	//static ILexer5 *LexerFactoryJulia() {
+	static ILexer *LexerFactoryJulia() {
 		return new LexerJulia();
 	}
 };
